packages/cli/validate-macos-version.ts

```diff
@@ -1,27 +1,36 @@
 import os from "node:os";
 
 /**
- * Validates that the current macOS version supports workerd.
- * Throws an error if running on macOS below version 13.5.
+ * Validates macOS version compatibility with workerd.
+ * @param shouldThrow - If true, throws an error on unsupported versions. If false, logs a warning.
  */
-export function validateMacOSVersion(): void {
+export function validateMacOSVersion(shouldThrow: boolean = true): void {
```

Comment:
What about
- using a more explicit name, i.e. `ensureMinimumMacOsVersion`
- Taking `{shouldThrow: boolean}` as an argument. IMO the argument should also be mandatary because how one would guess the default value?
- Document the CI behavior but IMO it should be moved to the caller if possible because nothing in the name nor doc indicates a different behavior in CI
- Document that this is a no-op outside of the darwin platform

Reply:
`ensureMinimumMacOsVersion` I'm afraid I disagree with this name change. This function does not ensure that the version is the minimum because if `{shouldThrow: false}` then it happily allows any version.

I could be convinced that `validateMacOSVersion` is not accurate either. So I propose `checkMacOSVersion()` instead.